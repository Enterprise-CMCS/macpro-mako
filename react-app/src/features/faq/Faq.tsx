import { useCallback, useEffect, useState } from "react";
import { Navigate, useParams } from "react-router";

import {
  Accordion,
  AccordionContent,
  AccordionItem,
  AccordionTrigger,
  Button,
  CardWithTopBorder,
  SubNavHeader,
} from "@/components";
import { useFeatureFlag } from "@/hooks/useFeatureFlag";

import { helpDeskContact, oneMACFAQContent } from "./content/oneMACFAQContent";

export const Faq = () => {
  const isFAQHidden = useFeatureFlag("TOGGLE_FAQ");

  const { id } = useParams<{ id: string }>();

  const [openItems, setOpenItems] = useState<string[]>([]);

  const expandAll = useCallback(() => {
    const allIds = [];
    oneMACFAQContent.forEach(({ qanda }) => {
      qanda.forEach(({ anchorText }) => allIds.push(anchorText));
    });
    setOpenItems(allIds);
  }, [setOpenItems]);

  useEffect(() => {
    if (id) {
      const element = document.getElementById(id);
      if (element) {
        setOpenItems([id]);
        window.scrollTo({
          top: element.offsetTop,
          behavior: "smooth",
        });
      }
    }
  }, [id]);

  // Get the flag value for hiding the MMDL banner.
  const isBannerHidden = useFeatureFlag("UAT_HIDE_MMDL_BANNER");
  const isUpdateNewLabel = useFeatureFlag("UPGRADE_NEW_LABEL");
  const isChipSpaDetailsEnabled = useFeatureFlag("CHIP_SPA_DETAILS");

  const chipSpaFAQ = isChipSpaDetailsEnabled
    ? {
        anchorText: "chip-spa-attachments",
        question: "What are the attachments for a CHIP SPA?",
        label: null,
        labelColor: null,
        answerJSX: (
          <>
            <p className="font-bold">CHIP SPA Attachment Types:</p>
            <p className="ml-12">Note: “*” indicates a required attachment.</p>
            <table className="faq-table ml-12 border-collapse border border-gray-300 w-full ">
              <tbody>
                <tr>
                  <th className="border border-gray-300 px-4 py-2">Attachment Name</th>
                  <th className="border border-gray-300 px-4 py-2">Description</th>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2">Current State Plan*</td>
                  <td className="border border-gray-300 px-4 py-2">
                    Current version of the CHIP state plan that details how the State operates its
                    CHIP program
                  </td>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2 whitespace-nowrap">
                    Amended State Plan Language*
                  </td>
                  <td className="border border-gray-300 px-4 py-2">
                    Track changes to <span className="underline">only</span> the currently approved
                    CHIP state plan pages that the State is proposing to amend
                  </td>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2">Cover Letter*</td>
                  <td className="border border-gray-300 px-4 py-2">
                    Cover letter to CMS with an authorized signature that outlines the purpose of
                    the CHIP SPA submission
                  </td>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2">Budget Documents</td>
                  <td className="border border-gray-300 px-4 py-2">
                    Updated 1-year budget if applicable of the State's planned expenditures if the
                    CHIP SPA submission has a significant impact on the approved budget
                  </td>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2">Public Notice</td>
                  <td className="border border-gray-300 px-4 py-2">
                    Process used by the State if applicable to accomplish involvement of the public
                    that occurred specifically for this CHIP SPA submission
                  </td>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2">Tribal Consultation</td>
                  <td className="border border-gray-300 px-4 py-2">
                    Consultation process with Indian Tribes if applicable that occurred specifically
                    for this CHIP SPA submission
                  </td>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2">Other</td>
                  <td className="border border-gray-300 px-4 py-2">
                    Other supporting document(s) needed to process the CHIP SPA submission
                  </td>
                </tr>
              </tbody>
            </table>

            <p className="font-bold mt-12">CHIP Eligibility SPA Attachment Types:</p>
            <p className="ml-12">Note: “*” indicates a required attachment.</p>
            <table className="faq-eligibility-table ml-12 border-collapse border border-gray-300 w-full ">
              <tbody>
                <tr>
                  <th className="border border-gray-300 px-4 py-2">Attachment Name</th>
                  <th className="border border-gray-300 px-4 py-2">Description</th>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2 whitespace-nowrap">
                    CHIP Eligibility Template*
                  </td>
                  <td className="border border-gray-300 px-4 py-2">
                    Amendment to the eligibility section of the CHIP state plan using a PDF template
                    repository, including statutory and regulatory background, required information,
                    and minimum review criteria
                  </td>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2">Cover Letter*</td>
                  <td className="border border-gray-300 px-4 py-2">
                    Cover letter to CMS with an authorized signature that outlines the purpose of
                    the CHIP SPA submission
                  </td>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2">Current State Plan</td>
                  <td className="border border-gray-300 px-4 py-2">
                    Current version of the CHIP state plan that details how the State operates its
                    CHIP program
                  </td>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2 whitespace-nowrap">
                    Amended State Plan Language
                  </td>
                  <td className="border border-gray-300 px-4 py-2">
                    Track changes to <span className="underline">only</span> the currently approved
                    CHIP state plan pages that the State is proposing to amend
                  </td>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2">Budget Documents</td>
                  <td className="border border-gray-300 px-4 py-2">
                    Updated 1-year budget if applicable of the State's planned expenditures if the
                    CHIP SPA submission has a significant impact on the approved budget
                  </td>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2">Public Notice</td>
                  <td className="border border-gray-300 px-4 py-2">
                    Process used by the State if applicable to accomplish involvement of the public
                    that occurred specifically for this CHIP SPA submission
                  </td>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2">Tribal Consultation</td>
                  <td className="border border-gray-300 px-4 py-2">
                    Consultation process with Indian Tribes if applicable that occurred specifically
                    for this CHIP SPA submission
                  </td>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2">Other</td>
                  <td className="border border-gray-300 px-4 py-2">
                    Other supporting document(s) needed to process the CHIP Eligibility SPA
                    submission
                  </td>
                </tr>
              </tbody>
            </table>
          </>
        ),
      }
    : {
        anchorText: "chip-spa-attachments",
        question: "What are the attachments for a CHIP SPA?",
        answerJSX: (
          <>
            <p>Note: “*” indicates a required attachment.</p>
            <table className="faq-table border-collapse border border-gray-300 w-full ">
              <tbody>
                <tr>
                  <th className="border border-gray-300 px-4 py-2">Attachment Name</th>
                  <th className="border border-gray-300 px-4 py-2">Description</th>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2">Current State Plan*</td>
                  <td className="border border-gray-300 px-4 py-2">
                    Current version of the CHIP state plan that details how the State operates its
                    CHIP program
                  </td>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2">Amended State Plan Language*</td>
                  <td className="border border-gray-300 px-4 py-2">
                    Redline version of proposed changes to the existing CHIP state plan pages. State
                    to provide a redline version and a clean version of the CHIP state plan pages
                    being amended.
                  </td>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2">Cover Letter*</td>
                  <td className="border border-gray-300 px-4 py-2">
                    Cover letter to CMS with an authorized signature that outlines the purpose of
                    the CHIP SPA submission
                  </td>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2">Budget Documents</td>
                  <td className="border border-gray-300 px-4 py-2">
                    Updated 1-year budget if applicable of the State's planned expenditures if the
                    CHIP SPA submission has a significant impact on the approved budget
                  </td>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2">Public Notice</td>
                  <td className="border border-gray-300 px-4 py-2">
                    Process used by the State if applicable to accomplish involvement of the public
                    that occurred specifically for this CHIP SPA submission
                  </td>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2">Tribal Consultation</td>
                  <td className="border border-gray-300 px-4 py-2">
                    Consultation process with Indian Tribes if applicable that occurred specifically
                    for this CHIP SPA submission
                  </td>
                </tr>
                <tr>
                  <td className="border border-gray-300 px-4 py-2">Other</td>
                  <td className="border border-gray-300 px-4 py-2">
                    Other document(s) needed to process the CHIP SPA submission
                  </td>
                </tr>
              </tbody>
            </table>
          </>
        ),
      };

  const anchorsToHide = [
    "spa-admendments",
    "abp-spa-templates",
    "abp-implementation-guides-spa",
    "mpc-spa-templates",
    "mpc-spa-implementation-guides",
    "chip-spa-templates",
    "chip-spa-implentation-guides",
  ];

  const filteredFAQContent = oneMACFAQContent.map((section) => {
    if (section.sectionTitle === "State Plan Amendments (SPAs)") {
      const filteredQanda = isBannerHidden
        ? section.qanda.filter((qa) => !anchorsToHide.includes(qa.anchorText))
        : section.qanda;

      const existingQanda = filteredQanda.filter((qa) => qa.anchorText !== "chip-spa-attachments");

      // Find the correct index for chip-spa-attachments to appear before RAI question
      const raiIndex = existingQanda.findIndex(
        (qa) => qa.anchorText === "chip-spa-rai-attachments",
      );

      const updatedQanda = [
        ...existingQanda.slice(0, raiIndex),
        chipSpaFAQ,
        ...existingQanda.slice(raiIndex),
      ];

      return { ...section, qanda: updatedQanda };
    }
    return section;
  });

  if (isFAQHidden) {
    return <Navigate to="/" replace />;
  }
  return (
    <>
      <SubNavHeader>
        <h1 className="text-xl font-medium">Frequently Asked Questions</h1>
      </SubNavHeader>
      <section className="block md:flex md:flex-row max-w-screen-xl m-auto px-4 lg:px-8 pt-8 gap-10">
        <div className="flex-1">
          <article className="mb-8">
            {/* BUTTON */}
            <Button
              onClick={expandAll}
              variant="outline"
              data-testid="expand-all"
              className="w-full xs:w-fit hover:bg-transparent mb-5"
            >
              Expand all to search with CTRL + F
            </Button>

            {/* FAQ */}
            <Accordion type="multiple" value={openItems} onValueChange={setOpenItems}>
              {filteredFAQContent.map(({ sectionTitle, qanda }) => (
                <article key={sectionTitle} className="mb-8">
                  <h2 className="text-2xl mb-4 text-primary">{sectionTitle}</h2>
                  {qanda.map(({ anchorText, answerJSX, question, label, labelColor }) => (
                    <AccordionItem
                      value={anchorText}
                      id={anchorText}
                      key={anchorText}
                      data-testid={anchorText}
                    >
                      <AccordionTrigger className="text-left">
                        <div className="flex items-center gap-2">
                          {isUpdateNewLabel && label && (
                            <span
                              className={`text-white text-xs font-semibold px-2 py-0.5 rounded no-underline hover:no-underline ${
                                labelColor === "green"
                                  ? "bg-green-600"
                                  : labelColor === "blue"
                                    ? "bg-blue-600"
                                    : "bg-gray-500"
                              }`}
                            >
                              {label}
                            </span>
                          )}
                          <span className="hover:underline">{question}</span>
                        </div>
                      </AccordionTrigger>
                      <AccordionContent>{answerJSX}</AccordionContent>
                    </AccordionItem>
                  ))}
                </article>
              ))}
            </Accordion>
          </article>
        </div>
        <div>
          <CardWithTopBorder className="sticky top-5">
            <div className="p-4">
              <h3 className="text-lg text-bold mb-4">OneMAC Help Desk Contact Info</h3>
              <div>
                <b>Phone Number</b>
                <p className="mb-4 text-primary">
                  <a className="underline" href={`tel:${helpDeskContact.phone}`}>
                    {helpDeskContact.phone}
                  </a>
                </p>
                <b>Email</b>
                <p className="text-primary">
                  <a className="underline" href={`mailto:${helpDeskContact.email}`}>
                    {helpDeskContact.email}
                  </a>
                </p>
              </div>
            </div>
          </CardWithTopBorder>
        </div>
      </section>
    </>
  );
};
