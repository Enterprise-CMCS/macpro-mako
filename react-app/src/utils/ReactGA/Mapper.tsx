export function mapSubmissionTypeBasedOnActionFormTitle(actionFormTitle) {
  let mappedTitle;
  if (actionFormTitle.includes("CHIP SPA Details")) {
    mappedTitle = "chip spa";
  } else if (actionFormTitle.includes("Medicaid SPA Details")) {
    mappedTitle = "medicaid spa";
  } else if (actionFormTitle.includes("Temporary Extension Request Details")) {
    mappedTitle = "temporary extension";
  } else if (
    actionFormTitle.includes("1915(b)(4) FFS Selective Contracting Initial Waiver Details")
  ) {
    mappedTitle = "1915b(4) initial waiver";
  } else if (
    actionFormTitle.includes("1915(b)(4) FFS Selective Contracting Renewal Waiver Details")
  ) {
    mappedTitle = "1915b(4) waiver renewal";
  } else if (
    actionFormTitle.includes("1915(b)(4) FFS Selective Contracting Waiver Amendment Details")
  ) {
    mappedTitle = "1915b(4) waiver amendment";
  } else if (actionFormTitle.includes("1915(b) Comprehensive (Capitated) Initial Waiver Details")) {
    mappedTitle = "1915b capitated inital";
  } else if (actionFormTitle.includes("1915(b) Comprehensive (Capitated) Renewal Waiver Details")) {
    mappedTitle = "1915b capitated renewal";
  } else if (
    actionFormTitle.includes("1915(b) Comprehensive (Capitated) Waiver Amendment Details")
  ) {
    mappedTitle = "1915b capitated amendment";
  } else if (actionFormTitle.includes("1915(c) Appendix K Amendment Details")) {
    mappedTitle = "1915c app-k";
  }
  return mappedTitle;
}
