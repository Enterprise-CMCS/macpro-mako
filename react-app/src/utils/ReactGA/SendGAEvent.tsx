export const sendGAEvent = (eventName, eventObject) => { 
  if (eventObject && typeof window.gtag == "function" ) {
    window.gtag("event", eventName, eventObject)
  } else {
    window.gtag("event", eventName);
  }
}