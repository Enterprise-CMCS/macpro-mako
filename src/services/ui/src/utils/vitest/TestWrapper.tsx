import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { ReactLocation, Router } from "@tanstack/react-location";

const queryClient = new QueryClient();
const location = new ReactLocation();

export const TestWrapper = ({ children }: any) => (
  <QueryClientProvider client={queryClient}>
    <Router location={location} routes={[]}>
      {children}
    </Router>
  </QueryClientProvider>
);
