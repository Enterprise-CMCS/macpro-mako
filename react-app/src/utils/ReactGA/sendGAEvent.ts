import ReactGA from "react-ga4";

export function sendGAEvent(eventTitle: string, userRoles: string, userState: string) {
  if (userState) {
    ReactGA.event({
      action: eventTitle,
      category: userState,
      user_role: userRoles,
    });
  } else {
    ReactGA.event({
      action: eventTitle,
      user_role: userRoles,
    });
  }
}
