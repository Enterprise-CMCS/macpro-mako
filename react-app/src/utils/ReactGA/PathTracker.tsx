import { useEffect, useRef, useState } from "react";
import { isCmsUser, isStateUser } from "shared-utils";

import { sendGAEvent } from "./SendGAEvent";

type PathTrackerProps = {
  userRole: "state" | "cms";
  children: React.ReactNode;
};

export default function PathTracker({ children }: PathTrackerProps) {
  const prevPathRef = useRef<string>(window.location.pathname);
  const startTimeRef = useRef<number>(Date.now());

  const [role, setRole] = useState<"state" | "cms" | null>(null);

  useEffect(() => {
    try {
      let detectedRole;
      if (isStateUser && !isCmsUser) {
        detectedRole = "state";
      } else if (isCmsUser && !isStateUser) {
        detectedRole = "cms";
      } else if (isStateUser && isCmsUser) {
        detectedRole = "state";
      }
      setRole(detectedRole);
    } catch (e) {
      console.error("error obtaining user roles for path tracking", e);
      setRole("state");
    }
  }, []);

  useEffect(() => {
    if (!role) return; // wait for role to be detected before tracking

    const sendPageView = (path: string) => {
      sendGAEvent("page_view", {
        page_path: path,
        referrer: prevPathRef.current || "",
        user_role: role,
      });
    };

    const sendPageDuration = (path: string, startTs: number) => {
      const now = Date.now();
      const deltaMs = now - startTs;
      const timeOnPageSec = Math.round(deltaMs / 1000);
      sendGAEvent("page_duration", {
        page_path: path,
        user_role: role,
        time_on_page_sec: timeOnPageSec,
      });
    };

    sendPageView(window.location.pathname);

    const onRouteChange = () => {
      const newPath = window.location.pathname;
      const oldPath = prevPathRef.current;
      if (newPath !== oldPath) {
        sendPageDuration(oldPath, startTimeRef.current);
        sendPageView(newPath);
        prevPathRef.current = newPath;
        startTimeRef.current = Date.now();
      }
    };

    const origPush = window.history.pushState;
    const origReplace = window.history.replaceState;

    window.history.pushState = function (...args) {
      origPush.apply(this, args);
      onRouteChange();
    };
    window.history.replaceState = function (...args) {
      origReplace.apply(this, args);
      onRouteChange();
    };

    window.addEventListener("popstate", onRouteChange);

    return () => {
      sendPageDuration(prevPathRef.current, startTimeRef.current);
      window.history.pushState = origPush;
      window.history.replaceState = origReplace;
      window.removeEventListener("popstate", onRouteChange);
    };
  }, [role]);

  return <>{children}</>;
}
