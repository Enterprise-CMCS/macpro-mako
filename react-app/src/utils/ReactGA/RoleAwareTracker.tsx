// src/components/RoleAwareTracker.tsx
import { useEffect, useState } from "react";
import { isCmsUser, isStateUser } from "shared-utils";
import { useGetUser } from "@/api";
const { data: userObj } = useGetUser();
import PathTracker from "./PathTracker"; // adjust path if needed

type UserRole = "cms" | "state";

function setGAUserRoleOnce(role: string) {
  if (typeof window === "undefined" || !window.gtag) return;

  // Prevent setting it multiple times
  if (window.__gaUserRoleSet) return;
  window.gtag("set", { user_role: role });
  window.__gaUserRoleSet = true; // Custom flag on window to track setting
}

/**
 * RoleAwareTracker
 *
 * 1) Reads the Cognito userData blob from Local Storage.
 * 2) Extracts either "custom:cms-roles" or "custom:ismemberof".
 * 3) Maps that attribute string into either "state" or "cms".
 */
export function RoleAwareTracker({ children }: { children: React.ReactNode }) {
  const [userRole, setUserRole] = useState<UserRole | null>(null);
  useEffect(() => {
    // 1) Find the Local Storage key that contains "CognitoIdentityServiceProvider.*.userData"
    const allKeys = Object.keys(localStorage);
    const userDataKey = allKeys.find(
      (k) => k.endsWith(".userData") && k.startsWith("CognitoIdentityServiceProvider."),
    );

    if (!userDataKey) {
      // No userData in localStorage or not logged in yet.t.
      return;
    }

    try {
      let role: UserRole | null = null;
      if (isCmsUser(userObj.user)) {
        role = "cms";
      } else if (isStateUser(userObj.user)) {
        role = "state";
      } 
      if (role) {
        setUserRole(role);
        setGAUserRoleOnce(role);
      }
    } catch (e) {
      console.log("error obtaining user roles for path tracking", e);
      // If parsing fails for or roles dont exist yet (initial page load before login)
      setUserRole("state");
    }
  }, []);
  return <PathTracker userRole={userRole}>{children}</PathTracker>;
}
