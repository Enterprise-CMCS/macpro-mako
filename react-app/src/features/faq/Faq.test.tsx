import { fireEvent, render, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";

import { sendGAEvent } from "@/utils/ReactGA/SendGAEvent";

import { Faq } from "./Faq";
vi.mock("react-router", async () => ({
  ...(await vi.importActual<Record<string, unknown>>("react-router")),
  useParams: vi.fn().mockReturnValue({ id: "q1" }),
}));
vi.mock("@/utils/ReactGA/SendGAEvent", () => ({
  sendGAEvent: vi.fn(),
}));
vi.mock("./faqs", () => ({
  oneMACFAQContent: [
    {
      sectionTitle: "Section 1",
      qanda: [
        { anchorText: "q1", question: "What is FAQ 1?", answerJSX: <p>Answer 1</p> },
        { anchorText: "q2", question: "What is FAQ 2?", answerJSX: <p>Answer 2</p> },
      ],
    },
  ],
}));

it("should expand all FAQ items when the 'Expand all' button is clicked and send GA event", () => {
  render(<Faq />);

  const expandButton = screen.getByTestId("expand-all");

  expect(screen.queryByText("Answer 2")).toBeNull();

  fireEvent.click(expandButton);

  expect(screen.getByText("Answer 2")).toBeInTheDocument();

  // GA event called
  expect(sendGAEvent).toHaveBeenCalledWith("support_click_general_expand-all", {
    event_category: "Support",
    event_label: "Expand All",
  });
});

it("should fire GA event when help desk phone is clicked", () => {
  render(<Faq />);

  const phoneLink = screen.getByText("(833) 228-2540");

  fireEvent.click(phoneLink);

  expect(sendGAEvent).toHaveBeenCalledWith("support_contact_phone");
});

it("should fire GA event when help desk email is clicked", () => {
  render(<Faq />);

  const emailLink = screen.getByText("OneMAC_Helpdesk@cms.hhs.gov");

  fireEvent.click(emailLink);

  expect(sendGAEvent).toHaveBeenCalledWith("support_contact_email");
});

describe("Faq", () => {
  it("should set open items correctly when id param is passed", async () => {
    const scrollToMock = vi.fn();
    global.scrollTo = scrollToMock;

    render(<Faq />);

    expect(screen.getByTestId("q1")).toBeInTheDocument();
    expect(scrollToMock).toHaveBeenCalled();
    expect(scrollToMock).toHaveBeenCalledWith({
      top: expect.any(Number),
      behavior: "smooth",
    });
  });

  it("should display the items expected", async () => {
    const scrollToMock = vi.fn();
    global.scrollTo = scrollToMock;

    const { asFragment } = render(<Faq />);

    expect(asFragment()).toMatchSnapshot();
  });
});
