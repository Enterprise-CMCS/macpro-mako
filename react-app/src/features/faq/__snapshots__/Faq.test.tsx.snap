// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Faq > should display the items expected 1`] = `
<DocumentFragment>
  <div
    class="bg-sky-100"
    data-testid="sub-nav-header"
  >
    <div
      class="max-w-screen-xl m-auto px-4 lg:px-8"
    >
      <div
        class="flex items-center"
      >
        <div
          class="flex align-middle py-4"
        >
          <h1
            class="text-xl font-medium"
          >
            Frequently Asked Questions
          </h1>
        </div>
      </div>
    </div>
  </div>
  <section
    class="block md:flex md:flex-row max-w-screen-xl m-auto px-4 lg:px-8 pt-8 gap-10"
  >
    <div
      class="flex-1"
    >
      <article
        class="mb-8"
      >
        <button
          aria-controls="faq-accordions"
          aria-expanded="false"
          class="inline-flex items-center justify-center rounded-md text-[16px] ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none border-2 border-primary text-primary font-bold disabled:opacity-50 h-10 px-4 py-2 w-full xs:w-fit hover:bg-transparent mb-5"
          data-testid="expand-all"
          id="expand-all-btn"
        >
          Expand all to search with CTRL + F
        </button>
        <div
          data-orientation="vertical"
          id="faq-accordions"
        >
          <article
            class="mb-8"
          >
            <h2
              class="text-2xl mb-4 text-primary"
            >
              Section 1
            </h2>
            <div
              class="border-b"
              data-orientation="vertical"
              data-state="open"
              data-testid="q1"
              id="q1"
            >
              <h3
                class="flex"
                data-orientation="vertical"
                data-state="open"
              >
                <button
                  aria-controls="radix-:rh:"
                  aria-expanded="true"
                  class="flex flex-1 items-center justify-between py-4 font-medium transition-all [&[data-state=open]>svg]:rotate-180 text-left"
                  data-orientation="vertical"
                  data-radix-collection-item=""
                  data-state="open"
                  id="radix-:rg:"
                  type="button"
                >
                  <div
                    class="flex items-center gap-2"
                  >
                    <span
                      class="hover:underline"
                    >
                      What is FAQ 1?
                    </span>
                  </div>
                  <svg
                    class="lucide lucide-chevron-down h-4 w-4 shrink-0 transition-transform duration-200"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m6 9 6 6 6-6"
                    />
                  </svg>
                </button>
              </h3>
              <div
                aria-labelledby="radix-:rg:"
                class="text-sm transition-all px-3"
                data-orientation="vertical"
                data-state="open"
                id="radix-:rh:"
                role="region"
                style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
              >
                <div
                  class="pb-4 pt-0"
                >
                  <p>
                    Answer 1
                  </p>
                </div>
              </div>
            </div>
            <div
              class="border-b"
              data-orientation="vertical"
              data-state="closed"
              data-testid="q2"
              id="q2"
            >
              <h3
                class="flex"
                data-orientation="vertical"
                data-state="closed"
              >
                <button
                  aria-controls="radix-:rj:"
                  aria-expanded="false"
                  class="flex flex-1 items-center justify-between py-4 font-medium transition-all [&[data-state=open]>svg]:rotate-180 text-left"
                  data-orientation="vertical"
                  data-radix-collection-item=""
                  data-state="closed"
                  id="radix-:ri:"
                  type="button"
                >
                  <div
                    class="flex items-center gap-2"
                  >
                    <span
                      class="hover:underline"
                    >
                      What is FAQ 2?
                    </span>
                  </div>
                  <svg
                    class="lucide lucide-chevron-down h-4 w-4 shrink-0 transition-transform duration-200"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m6 9 6 6 6-6"
                    />
                  </svg>
                </button>
              </h3>
              <div
                aria-labelledby="radix-:ri:"
                class="text-sm transition-all px-3"
                data-orientation="vertical"
                data-state="closed"
                hidden=""
                id="radix-:rj:"
                role="region"
                style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
              />
            </div>
          </article>
        </div>
      </article>
    </div>
    <div>
      <div
        class="my-6 flex flex-col sticky top-5"
      >
        <div
          class="h-2 shadow-lg"
          style="border-radius: 3px 3px 0px 0px; height: 8px;"
        />
        <div
          class="flex-1 border border-t-0 rounded-b-sm border-slate-300"
        >
          <div
            class="p-4"
          >
            <h3
              class="text-lg text-bold mb-4"
            >
              OneMAC Help Desk Contact Info
            </h3>
            <div>
              <b>
                Phone Number
              </b>
              <p
                class="mb-4 text-primary"
              >
                <a
                  class="underline"
                  href="tel:(833) 228-2540"
                >
                  (833) 228-2540
                </a>
              </p>
              <b>
                Email
              </b>
              <p
                class="text-primary"
              >
                <a
                  class="underline"
                  href="mailto:OneMAC_Helpdesk@cms.hhs.gov"
                >
                  OneMAC_Helpdesk@cms.hhs.gov
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
